lib/src/models/multireddit.dart

```diff
@@ -3,13 +3,406 @@
+  String _author;
+
+  /// The displayName given to the [Multireddit].
+  String get displayName => _data['display_name'];
+
+  /// The infoPath will be used to uniquely identify this multireddit.
+  String get infoPath => _infoPath ?? '/';
+  String _infoPath;
+
+  /// The visibility of this multireddit.
+  ///
+  /// Can be one of { hidden, private, public }.
+  /// If this information is not provided, will return null.
+  Visibility get visibility => Visibility.values.firstWhere(
+      (e) => e.toString() == ('Visibility.' + _data['visibility']),
+      orElse: () => null);
+
+  /// The visibility of this multireddit.
+  ///
+  /// Can be one of { hidden, private, public }.
+  /// If this information is not provided, will return null.
+  WeightingScheme get weightingScheme => WeightingScheme.values.firstWhere(
+      (e) => e.toString() == ('WeightingScheme.' + _data['weighting_scheme']),
+      orElse: () => null);
+
+  /// Does the currently authenticated [User] have the privilege to edit this
+  /// multireddit.
+  bool get canEdit => _data['can_edit'];
+
+  /// Does this multireddit require visitors to be over the age of 18.
+  bool get over18 => _data['over_18'];
+
   Multireddit.parse(Reddit reddit, Map data) : super(reddit) {
-    setData(this, data['data']);
+    _data = data['data'];
+    _author = _data['path']?.split('/')[_redditorNameInPathIndex];
+    _infoPath = _generateInfoPath(_data['name'], _author);
+  }
+
+  // Returns valid info_path for multireddit with name `name`.
+  static String _generateInfoPath(String name, String user) =>
+      apiPath['multireddit']
+          .replaceAll(_multiredditRegExp, name)
+          .replaceAll(_userRegExp, user);
+
+  // Converts input text into camelCase.
+  static String _convertToCamelCase(String str) {
+    if (str == null) {
+      return null;
+    }
+    final splitStrList = str.split(new RegExp(r'(\s|_)+'));
+    final strItr = splitStrList.iterator;
+    final strBuffer = new StringBuffer(strItr.toString());
+    while (strItr.moveNext()) {
+      strBuffer.write(strItr.toString().substring(0, 1).toUpperCase());
+      strBuffer.write(strItr.toString().substring(1));
+    }
+    return strBuffer.toString();
+  }
+
+  /// Returns a slug version of the [title].
+  static String _sluggify(String title) {
+    if (title == null) {
+      return null;
+    }
+    final RegExp _invalidRegExp = new RegExp(r'(\s|\W|_)+');
+    var titleScoped = title.replaceAll(_invalidRegExp, '_').trim();
+    if (titleScoped.length > 21) {
+      titleScoped = titleScoped.substring(21);
+      final lastWord = titleScoped.lastIndexOf('_');
+      //TODO(ckartik): Test this well.
+      if (lastWord > 0) {
+        titleScoped = titleScoped.substring(lastWord);
+      }
+    }
+    return titleScoped ?? '_';
+  }
+
+  /// Add a [Subreddit] to this [Multireddit].
+  ///
+  /// [subreddit] is the name of the [Subreddit] to be added to this [Multireddit].
```

Comment:
Use \`subreddit\` instead of [subreddit] for params (square braces signal that the text contained is a reference to a class, method, or property).

Reply:
done.